### ISSUE TITLE
synchronize error with paritydb

### ISSUE THREAD

**Issue by kogeler** [paritytech/polkadot#4604]
I can't synchronize the node from scratch using paritydb.
Restarts don't help.
Binary version is `0.9.14-80dfb926f4-x86_64-linux-gnu` (from GitHub releases).
OS: `Debian 10 (AMD64)`
CLI: `kusama-paritydb-archive --base-path /opt/kusama-paritydb-archive --name kusama-sync-0-kusama-paritydb-archive --chain kusama --listen-addr=/ip4/0.0.0.0/tcp/30337 --public-addr=/ip4/1.1.1.17/tcp/30337 --in-peers 25 --out-peers 25 --database paritydb-experimental --db-cache 512 --telemetry-url wss://a.b/submit/ 1 --pruning=archive --prometheus-external --prometheus-port 9619 --ws-port 9948 --ws-max-connections 100 --rpc-port 9937`

```
2021-12-25 07:59:18 ----------------------------
2021-12-25 07:59:18 This chain is not in any way
2021-12-25 07:59:18       endorsed by the
2021-12-25 07:59:18      KUSAMA FOUNDATION
2021-12-25 07:59:18 ----------------------------
2021-12-25 07:59:18 Parity Polkadot
2021-12-25 07:59:18 ✌  version 0.9.14-80dfb926f4-x86_64-linux-gnu
2021-12-25 07:59:18 ❤  by Parity Technologies <admin@parity.io>, 2017-2021
2021-12-25 07:59:18 📋 Chain specification: Kusama
2021-12-25 07:59:18 🏷 Node name: kusama-sync-0-kusama-paritydb-archive
2021-12-25 07:59:18 👤 Role: FULL
2021-12-25 07:59:18 💾 Database: ParityDb at /opt/kusama-paritydb-archive/chains/ksmcc3/paritydb/full
2021-12-25 07:59:18 ⛓  Native runtime: kusama-9140 (parity-kusama-0.tx8.au2)
2021-12-25 07:59:18 🏷 Local node identity is: 12D3KooWEY1iVNFMxj44RFGispvqzSzcwo5ab6wTXZbMNqa6rym8
2021-12-25 07:59:18 🔍 Discovered new external address for our node: /ip4/34.107.77.47/tcp/30337/p2p/12D3KooWEY1iVNFMxj44RFGispvqzSzcwo5ab6wTXZbMNqa6rym8
2021-12-25 07:59:18 📦 Highest known block at #6520511
2021-12-25 07:59:18 〽 Prometheus exporter started at 0.0.0.0:9619
2021-12-25 07:59:18 Listening for new connections on 127.0.0.1:9948.
2021-12-25 07:59:23 ⚙  Syncing, target=#10670981 (5 peers), best: #6520511 (0x6eba…05b8), finalized #6520320 (0x4d06…dd92), ⬇ 495.7kiB/s ⬆ 14.7kiB/s
====================
Version: 0.9.14-80dfb926f4-x86_64-linux-gnu
   0: sp_panic_handler::set::{{closure}}
   1: std::panicking::rust_panic_with_hook
             at rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:628:17
   2: std::panicking::begin_panic_handler::{{closure}}
             at rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:521:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys_common/backtrace.rs:141:18
   4: rust_begin_unwind
             at rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:517:5
   5: core::panicking::panic_fmt
             at rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/panicking.rs:101:14
   6: core::result::unwrap_failed
             at rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/result.rs:1617:5
   7: <sp_state_machine::ext::Ext<H,B> as sp_externalities::Externalities>::storage
   8: sp_io::storage::get_version_1
   9: <<sp_io::storage::HostFunctions as sp_wasm_interface::HostFunctions>::host_functions::ExtStorageGetVersion1 as sp_wasm_interface::Function>::execute
  10: sc_executor_wasmtime::imports::HostFuncHandler::into_extern::{{closure}}
  11: wasmtime::func::Func::invoke
  12: wasmtime::trampoline::func::stub_fn
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: wasmtime_runtime::traphandlers::catch_traps::call_closure
  22: wasmtime_setjmp
  23: sc_executor_wasmtime::runtime::perform_call
  24: <sc_executor_wasmtime::runtime::WasmtimeInstance as sc_executor_common::wasm_runtime::WasmInstance>::call
  25: sc_executor_common::wasm_runtime::WasmInstance::call_export
  26: sc_executor::native_executor::WasmExecutor::with_instance::{{closure}}
  27: sp_state_machine::execution::StateMachine<B,H,Exec>::execute_aux
  28: <kusama_runtime::RuntimeApiImpl<__SR_API_BLOCK__,RuntimeApiImplCall> as sp_api::Core<__SR_API_BLOCK__>>::Core_execute_block_runtime_api_impl
  29: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  30: <sc_finality_grandpa::import::GrandpaBlockImport<BE,Block,Client,SC> as sc_consensus::block_import::BlockImport<Block>>::import_block::{{closure}}
  31: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  32: <sc_consensus_babe::BabeBlockImport<Block,Client,Inner> as sc_consensus::block_import::BlockImport<Block>>::import_block::{{closure}}
  33: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  34: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  35: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  36: <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll
  37: <tracing_futures::Instrumented<T> as core::future::future::Future>::poll
  38: tokio::runtime::task::raw::poll
  39: std::sys_common::backtrace::__rust_begin_short_backtrace
  40: core::ops::function::FnOnce::call_once{{vtable.shim}}
  41: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/alloc/src/boxed.rs:1636:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/alloc/src/boxed.rs:1636:9
      std::sys::unix::thread::Thread::new::thread_start
             at rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys/unix/thread.rs:106:17
  42: start_thread
  43: clone
Thread 'tokio-runtime-worker' panicked at 'Externalities not allowed to fail within runtime: "Trie lookup error: Database missing expected key: 0x6149cdc46c350cde3e7fa5809aa235985dc12cf7e55d153990aace6fd6179e15"', /usr/local/cargo/git/checkouts/substrate-7e08433d4c370a21/de6033a/primitives/state-machine/src/ext.rs:187
This is a bug. Please report it at:
        https://github.com/paritytech/polkadot/issues/new
kusama-paritydb-archive.service: Main process exited, code=exited, status=1/FAILURE
kusama-paritydb-archive.service: Failed with result 'exit-code'.
kusama-paritydb-archive.service: Service RestartSec=1min 30s expired, scheduling restart.
kusama-paritydb-archive.service: Scheduled restart job, restart counter is at 291.

```

**Comment by bkchr** [comment]
Cc @arkpar

@kogeler can you backup the db?

**Comment by arkpar** [comment]
Can't reproduce with master. Looks like the same issue as https://github.com/paritytech/polkadot/issues/4169, which should be fixed in parity-db 0.3.4 / polkadot 0.9.14. Could it be that the database was initially synced with an earlier version of polkadot?

**Comment by kogeler** [comment]
@arkpar  Yes, it was started with the 0.9.12 version firstly. Thanks!
